pisek/jobs/cache.py

```diff
@@ -27,15 +27,30 @@
 SAVED_LAST_SIGNATURES = 5
 
 
-class YAMLObjectWithRequiredAttr(yaml.YAMLObject):
-    """Yaml object that has must have all attributes set."""
+class CacheInfo(yaml.YAMLObject):
+    """Object for cache metadata."""
 
-    @classmethod
-    def from_yaml(cls, loader, node):
-        return cls(**loader.construct_mapping(node, deep=True))
+    yaml_tag = "!Info"
 
+    def __init__(self, version: str = __version__) -> None:
+        self.version = version
 
-class CacheEntry(YAMLObjectWithRequiredAttr):
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}(version={self.version})"
+
+    @staticmethod
+    def read(f: TextIO) -> "CacheInfo":
+        tag = f.readline().removeprefix("- ").strip()
+        if tag != CacheInfo.yaml_tag:
+            return CacheInfo("?.?.?")
+        version = f.readline().removeprefix("  version:").strip()
+        return CacheInfo(version)
```

Comment:
Is it guaranteed that `CacheInfo` will be the first entry?

Reply:
For all new versions, yes.